addons/mail/static/src/views/form/form_compiler.js

```diff
@@ -0,0 +1,125 @@
+                break;
+        }
+    }
+    const chatterContainerXml = createElement("ChatterContainer");
+    chatterContainerXml.setAttribute("hasActivities", hasActivities);
+    chatterContainerXml.setAttribute("hasFollowers", hasFollowers);
+    chatterContainerXml.setAttribute("hasMessageList", hasMessageList);
+    chatterContainerXml.setAttribute("hasParentReloadOnAttachmentsChanged", hasParentReloadOnAttachmentsChanged);
+    chatterContainerXml.setAttribute("hasParentReloadOnFollowersUpdate", hasParentReloadOnFollowersUpdate);
+    chatterContainerXml.setAttribute("hasParentReloadOnMessagePosted", hasParentReloadOnMessagePosted);
+    chatterContainerXml.setAttribute("isAttachmentBoxVisibleInitially", isAttachmentBoxVisibleInitially);
+    chatterContainerXml.setAttribute("threadId", "props.record.resId or undefined");
+    chatterContainerXml.setAttribute("threadModel", "props.record.resModel");
+    chatterContainerXml.setAttribute("webRecord", "props.record");
+
+    const chatterContainerHookXml = createElement("div");
+    chatterContainerHookXml.classList.add("o_FormRenderer_chatterContainer");
+    append(chatterContainerHookXml, chatterContainerXml);
+    return chatterContainerHookXml;
+}
+
+function compileAttachmentPreview(node, params) {
+    const webClientViewAttachmentViewContainerHookXml = createElement("div");
+    webClientViewAttachmentViewContainerHookXml.classList.add('o_attachment_preview');
+    const webClientViewAttachmentViewContainerXml = createElement("WebClientViewAttachmentViewContainer");
+    webClientViewAttachmentViewContainerXml.setAttribute("threadId", "props.record.resId or undefined");
+    webClientViewAttachmentViewContainerXml.setAttribute("threadModel", "props.record.resModel");
+    append(webClientViewAttachmentViewContainerHookXml, webClientViewAttachmentViewContainerXml);
+    return webClientViewAttachmentViewContainerHookXml;
+}
+
+registry.category("form_compilers").add("chatter_compiler", {
+    selector: "div.oe_chatter",
+    fn: compileChatter,
+});
+
+registry.category("form_compilers").add("attachment_preview_compiler", {
+    selector: "div.o_attachment_preview",
+    fn: compileAttachmentPreview,
+});
+
+patch(FormCompiler.prototype, 'mail', {
+    compile() {
+        // TODO no chatter if in dialog?
+        const res = this._super(...arguments);
+        const chatterContainerHookXml = res.querySelector('.o_FormRenderer_chatterContainer');
+        if (!chatterContainerHookXml) {
+            return res; // no chatter, keep the result as it is
+        }
+        const chatterContainerXml = chatterContainerHookXml.querySelector('ChatterContainer');
+        if (chatterContainerHookXml.parentNode.classList.contains('o_form_sheet')) {
+            chatterContainerXml.setAttribute("hasExternalBorder", 'true');
+            chatterContainerXml.setAttribute("hasMessageListScrollAdjust", 'false');
```

Comment:
usage of a helper to set multiple attributes from an object would really help readability of this entire file IMO. You can add one in `@web/core/utils/xml.js`

Reply:
createElement already does kinda that